### DISCUSSION TITLE
Script seems to be adding duplicate calendar entries

### DISCUSSION THREAD

**Discussion by corhof** [derekantrican/GAS-ICS-Sync#451] [Q&A]
I have the script referencing 4 ical links.  On one link, it imported all events in the past but did not import any future events.  On the other links, it’s importing duplicates…as many as 10 times.  Look at the screenshot below to see what I’m talking about.  
![IMG_5851](https://github.com/user-attachments/assets/32a27c44-8787-42e0-aafe-7f9c4d0e40f8)

**Comment by jonas0b1011001** [comment]
Can you share a log from https://script.google.com/home/executions

**Comment by corhof** [comment]
Here's a screenshot
<img width="1420" alt="Screenshot 2024-08-19 at 9 13 47 AM" src="https://github.com/user-attachments/assets/3a5b18ba-8e73-4b85-b792-b9d707925e54">

**Comment by jonas0b1011001** [comment]
Two things:
1. Looks like a common issue with non persistent event ids. This causes the script to sync all events as new on every run.
2. You seem to sync quite a huge amount of events. As the script has limited runtime due to google restriction, it times out and does not delete the events from the previous runs.

Your options:
- limit the amount of events to sync (e.g. use onlyFutureEvents)
- see if the version linked in #283 resolves the issue

**Comment by corhof** [comment]
That worked!  Thank you.
